packages/engine-test-utils/src/__tests__/pods-core/GraphvizPod.spec.ts

```diff
@@ -0,0 +1,175 @@
+          engine,
+          vaults,
+          wsRoot,
+          config: {
+            dest: exportDest,
+            includeBody: true,
+            includeStubs: true,
+            includeHierarchy: true,
+            includeLinks: false,
+          },
+        });
+
+        // check that graphviz file is created
+        let [expectedFiles, actualFiles] = FileTestUtils.cmpFiles(exportDest, [
+          "graphviz.dot",
+        ]);
+        expect(expectedFiles).toEqual(actualFiles);
+
+        // check contents of graphviz file
+        const dotFile = fs.readFileSync(path.join(exportDest, "graphviz.dot"), {
+          encoding: "utf8",
+        });
+
+        // TODO: IDs of files switch between test runs, which causes the snapshot
+        // to fail when it shouldn't
+        // expect(dotFile).toMatchSnapshot("graphviz contents");
+
+        // Check for:
+        // 1. start of file -> "graph {"
+        // 2. existence of a note -> "note_"
+        // 3. a labeled note -> "[label=\""
+        // 4. a connection -> "--"
+        // 5. end of graph -> "}"
+        await checkString(dotFile, "graph {", "note_", '[label="', "--", "}");
+      },
+      { expect, preSetupHook: setupBasic }
+    );
+  });
+
+  test("include hierarchical connections only", async () => {
+    await runEngineTestV5(
+      async ({ engine, vaults, wsRoot }) => {
+        const pod = new GraphvizExportPod();
+        engine.config.useFMTitle = true;
+        await pod.execute({
+          engine,
+          vaults,
+          wsRoot,
+          config: {
+            dest: exportDest,
+            includeBody: true,
+            includeStubs: true,
+            includeHierarchy: true,
+            includeLinks: false,
+          } as GraphvizExportConfig,
+        });
+
+        // check contents of graphviz file
+        const dotFile = fs.readFileSync(path.join(exportDest, "graphviz.dot"), {
+          encoding: "utf8",
+        });
+
+        // check for hierarchical-specific elements
+        await checkString(dotFile, "--");
+        await checkNotInString(dotFile, "dotted");
+      },
+      { expect, preSetupHook: setupBasic }
+    );
+  });
+
+  test("include link connections only", async () => {
+    await runEngineTestV5(
+      async ({ engine, vaults, wsRoot }) => {
+        const pod = new GraphvizExportPod();
+        engine.config.useFMTitle = true;
+        await pod.execute({
+          engine,
+          vaults,
+          wsRoot,
+          config: {
+            dest: exportDest,
+            includeBody: true,
+            includeStubs: true,
+            includeHierarchy: false,
+            includeLinks: true,
+          } as GraphvizExportConfig,
+        });
+
+        // check contents of graphviz file
+        const foo = fs.readFileSync(path.join(exportDest, "graphviz.dot"), {
+          encoding: "utf8",
+        });
+
+        // check for link-specific elements
+        await checkString(foo, "--", "dotted");
```

Comment:
this looks same for hierarchy test. can we do two checks here? one for something that is only available in edges and one that is only available for hierarchies (the latter should evaluate to false)

Reply:
The current issue with this is the fact that linked notes are currently notated by a "dotted" line, whereas hierarchical connections are noted as a normal solid line. The "dotted" style is added on top of the base connection style, but nothing is added for hierarchical connections at the moment. I can look into a better of way of denoting the two so the difference is clearer

Reply:
in that case can we check for the non-existence of the "dotted" line?